plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1115,10 +1115,10 @@ func (c *PodAffinityChecker) getMatchingAntiAffinityTerms(pod *v1.Pod, allPods [
 
 // Checks if scheduling the pod onto this node would break any anti-affinity
 // rules indicated by the existing pods.
-func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) bool {
+func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, algorithm.PredicateFailureReason) {
```

Comment:
I see why you are returning `algorithm.PredicateFailureReason` in this function, but the failure reason is always `ErrExistingPodsAntiAffinityRulesNotMatch`, unless there is an error. It would be better to return an `error` instead of `algorithm.PredicateFailureReason` and then in `InterPodAffinityMatches` function you check the error. If it is nil, you add `ErrExistingPodsAntiAffinityRulesNotMatch` to the failure reasons. If it is not nil, you return the `error`. You should do this for `satisfiesPodsAffinityAntiAffinity` as well. This would be more compatible with the existing behavior and would also address a problem in the current code that errors occurred in `satisfiesPodsAffinityAntiAffinity` and `satisfiesExistingPodsAntiAffinity` are not trickled up.

Reply:
If we don't return algorithm.PredicateFailureReason in function satisfiesPodsAffinityAntiAffinity  , we have to check the specific error just as your comment, two cases:
a. check the error: PodAffinityRulesNotMatch
b. check the error: PodAntiAffinityRulesNotMatch 
if error is a or b, we add ErrPodsAffinityRulesNotMatch or ErrPodsAntiAffinityRulesNotMatch to algorithm.PredicateFailureReason in addition to ErrPodAffinityNotMatc and return nil. If got other errors we should return the error directly, right?